chpl/chpl-service/src/main/java/gov/healthit/chpl/validation/pendingListing/reviewer/FieldLengthReviewer.java

```diff
@@ -1,68 +1,99 @@
 package gov.healthit.chpl.validation.pendingListing.reviewer;
 
+import java.util.ArrayList;
+
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.context.MessageSource;
 import org.springframework.context.i18n.LocaleContextHolder;
 import org.springframework.context.support.DefaultMessageSourceResolvable;
 import org.springframework.stereotype.Component;
 
 import gov.healthit.chpl.dto.PendingCertifiedProductDTO;
+import gov.healthit.chpl.dto.PendingCertifiedProductTargetedUserDTO;
 import gov.healthit.chpl.util.ErrorMessageUtil;
 
 @Component("pendingFieldLengthReviewer")
 public class FieldLengthReviewer implements Reviewer {
-    @Autowired private ErrorMessageUtil msgUtil;
-    @Autowired private MessageSource messageSource;
+    @Autowired
+    private ErrorMessageUtil msgUtil;
+    @Autowired
+    private MessageSource messageSource;
+
+    private static String ERROR = "error";
+    private static String WARNING = "warning";
 
     @Override
     public void review(PendingCertifiedProductDTO listing) {
-        checkField(listing, listing.getCertificationEditionId(), "certificationEdition");
-        checkField(listing, listing.getAcbCertificationId(), "acbCertificationId");
-        checkField(listing, listing.getCertificationBodyId(), "certifyingAcb");
-        checkField(listing, listing.getUniqueId(), "uniqueCHPLId");
-        checkField(listing, listing.getDeveloperName(), "developerName");
-        checkField(listing, listing.getProductName(), "productName");
-        checkField(listing, listing.getProductVersion(), "productVersion");
+        checkField(listing, listing.getCertificationEditionId(), "certificationEdition", ERROR);
+        checkField(listing, listing.getAcbCertificationId(), "acbCertificationId", ERROR);
+        checkField(listing, listing.getCertificationBodyId(), "certifyingAcb", ERROR);
+        ArrayList<PendingCertifiedProductTargetedUserDTO> toRemove = new ArrayList<PendingCertifiedProductTargetedUserDTO>();
+        for (PendingCertifiedProductTargetedUserDTO tu : listing.getTargetedUsers()) {
+            checkField(listing, tu.getName(), "targetedUser", WARNING);
+            if (listing.getWarningMessages().contains(msgUtil.getMessage("listing.targetedUser.maxlength",
+                    String.valueOf(getMaxLength("maxLength.targetedUser")), tu.getName()))) {
```

Comment:
I am assuming this works when you upload something and that it was tested locally within the app... it just looks to me like it might not because the pending_certified_product_targeted_user table has a character limit of 300 on the targeted_user text so I'm confused as to how/when the targeted user is getting removed. It can't be put in the table if length > 300 characters so I feel like a db exception would have been thrown before this code is executed. I could be thinking about the workflow incorrectly so if it does work, that's fine. Otherwise I'm thinking there might also need to be a change to the pending table in the db to make the targeted user a text column that doesn't restrict length.